sdk/Sdk.Analyzers/RegistrationExpectedInASPNetIntegration.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class RegistrationExpectedInASPNetIntegration : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(DiagnosticDescriptors.CorrectRegistrationExpectedInAspNetIntegration);
+
+        private const string ExpectedRegistrationMethod = "ConfigureFunctionsWebApplication";
+        private const string AspNetExtensionAssemblyName = "Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore";
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+            context.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+        }
+
+        private static void AnalyzeMethod(SymbolAnalysisContext context)
+        {
+            var symbol = (IMethodSymbol)context.Symbol;
+
+            bool isAspNetAssembly = context.Compilation.ReferencedAssemblyNames.Any(assembly => assembly.Name.Equals(AspNetExtensionAssemblyName));
+
+            if (!isAspNetAssembly || !symbol.IsMainMethod())
+            {
+                return;
+            }
+
+            SyntaxReference syntaxReference = symbol.DeclaringSyntaxReferences.FirstOrDefault();
+            string code = syntaxReference.SyntaxTree.GetText().ToString();
+
+            if (!code.Contains(ExpectedRegistrationMethod))
```

Comment:
The string.Contains may fail if user has code like below.

```csharp
var host = new HostBuilder()
//.ConfigureFunctionsWebApplication()
 .Build();
```
Update one of the tests have this code.

Reply:
May be we should try something like this.

```csharp
SyntaxReference syntaxReference = symbol.DeclaringSyntaxReferences.FirstOrDefault();
var root = syntaxReference.SyntaxTree.GetRoot();

var methodCallExpressions = root.DescendantNodes().OfType<InvocationExpressionSyntax>();

bool methodInvocationPresent = methodCallExpressions.Any(invocation =>
    (invocation.Expression as MemberAccessExpressionSyntax)?.Name.Identifier.ValueText ==
    ExpectedRegistrationMethod);

if (methodInvocationPresent)
{
    return;
}
// Report diagnostic here

```

Reply:
Absolutely loved the suggestion here. Thank you!